database/scrapers/faq_script.ts

```diff
@@ -0,0 +1,190 @@
+  const [articlesResponse, questionsResponse, pivotTableResponse] =
+    await Promise.all([
+      fetch("https://admin.topwr.solvro.pl/items/FAQ_Types"),
+      fetch("https://admin.topwr.solvro.pl/items/FAQ"),
+      fetch("https://admin.topwr.solvro.pl/items/FAQ_Types_FAQ"),
+    ]);
+
+  if (!articlesResponse.ok) {
+    throw new Error(
+      `Failed to fetch articles - got response status code ${articlesResponse.status}`,
+    );
+  }
+  if (!questionsResponse.ok) {
+    throw new Error(
+      `Failed to fetch questions - got response status code ${questionsResponse.status}`,
+    );
+  }
+  if (!pivotTableResponse.ok) {
+    throw new Error(
+      `Failed to fetch pivot table - got response status code ${pivotTableResponse.status}`,
+    );
+  }
+  const [articlesResult, questionsResult, pivotTableResult] = await Promise.all(
+    [
+      articlesResponse.json() as Promise<GuideArticlesOld>,
+      questionsResponse.json() as Promise<GuideQuestionsOld>,
+      pivotTableResponse.json() as Promise<PivotTable>,
+    ],
+  );
+
+  for (const article of articlesResult.data) {
+    let createdAt: DateTime = DateTime.now();
+    let updatedAt = DateTime.fromMillis(0);
+
+    for (const pivot of pivotTableResult.data.filter(
+      (p) => p.FAQ_Types_id === article.id,
+    )) {
+      const question = questionsResult.data.find((q) => q.id === pivot.FAQ_id);
+
+      if (question === undefined) {
+        logger.warn(
+          `Pivot references missing question (ID=${pivot.FAQ_id}) for article ID=${article.id} ("${article.name}"). ` +
+            `This may be a data inconsistency in the source. Skipping this question...`,
+        );
+        continue;
+      }
+
+      const questionCreatedAt = DateTime.fromISO(question.date_created);
+      const questionUpdatedAt = DateTime.fromISO(question.date_updated);
+
+      if (questionCreatedAt < createdAt) {
+        createdAt = questionCreatedAt;
+      }
+
+      if (questionUpdatedAt > updatedAt) {
+        updatedAt = questionUpdatedAt;
+      }
+    }
+
+    const imagePath = await uploadImage(article.cover);
+
+    await GuideArticle.create({
+      id: article.id,
+      title: article.name,
+      shortDesc: article.short_description,
+      description: article.description ?? "",
+      imagePath,
+      createdAt,
+      updatedAt,
+    });
+  }
+
+  for (const pivot of pivotTableResult.data.filter(
+    (p) => p.FAQ_Types_id !== null,
+  )) {
+    const question = questionsResult.data.find((q) => q.id === pivot.FAQ_id);
+
+    if (question === undefined) {
+      logger.warn(
+        `Pivot references missing question (ID=${pivot.FAQ_id}) for article ID=${pivot.FAQ_Types_id}. ` +
+          `This may be a data inconsistency in the source. Skipping this question...`,
+      );
+      continue;
+    }
+
+    await GuideQuestion.create({
+      title: question.question,
+      answer: question.answer,
+      articleId: pivot.FAQ_Types_id ?? undefined,
+    });
```

Comment:
if TS does not understand that .filter() removed the possibility of pivot.FAQ_Types_id being null, then just use a regular if statement + continue in the loop instead of .filter(), instead of doing this funky shit

![image](https://github.com/user-attachments/assets/7c77cf25-1a54-4407-b9d9-f2e2d6e4287b)

might even be faster since it's not allocating another temporary array... but performance is kinda irrelevant in the context of these scrapers/migration scripts
as long as it completes in a reasonable amount of time, it's fine